<launch>
  <arg name="mav_name" default="quad"/>
  <arg name="fcu_url" default="udp://:14540@127.0.0.1:14557"/>
  <arg name="gcs_url" default="" />
  <arg name="tgt_system" default="1" />
  <arg name="tgt_component" default="1" />
  <arg name="command_input" default="2" />
  <arg name="gazebo_simulation" default="true" />
  <arg name="visualization" default="true"/>
  <arg name="log_output" default="screen" />
  <arg name="fcu_protocol" default="v2.0" />
  <arg name="respawn_mavros" default="false" />
  <arg name="gazebo_gui" default="true" />
  
  <node pkg="quadrotor_sim" type="quadrotor_sim" name="quadrotor_sim" output="screen">
      <rosparam command="load" file="$(find quadrotor_sim)/config/control_param.yaml"/>
      <rosparam command="load" file="$(find quadrotor_sim)/config/quad_param.yaml"/>
  		<param name="mav_name" type="string" value="$(arg mav_name)" />
          <remap from="command/bodyrate_command" to="/mavros/setpoint_raw/attitude"/>
          <param name="ctrl_mode" value="$(arg command_input)" />
          <param name="enable_sim" value="$(arg gazebo_simulation)" />
          <param name="enable_gazebo_state" value="true"/>
          <param name="maneuver_select" value="0"/>
  </node>

  <include file="$(find maneuvers)/launch/trajectory_launch.launch"/>

  <include file="$(find mavros)/launch/node.launch">
      <arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
      <arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

      <arg name="fcu_url" value="$(arg fcu_url)" />
      <arg name="gcs_url" value="$(arg gcs_url)" />
      <arg name="tgt_system" value="$(arg tgt_system)" />
      <arg name="tgt_component" value="$(arg tgt_component)" />
      <arg name="log_output" value="$(arg log_output)" />
      <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
      <arg name="respawn_mavros" default="$(arg respawn_mavros)" />
  </include>

  <include file="$(find px4)/launch/posix_sitl.launch">
      <arg name="vehicle" value="$(arg mav_name)"/>
      <arg name="world"   value="$(find quadrotor_sim)/worlds/basic_world.world"/>
      <arg name="sdf"     value="$(find quadrotor_sim)/models/quad/quad.sdf"/>
  </include>

</launch>
